### PR TITLE
Simplify tools installation & pipelines

### PR DESCRIPTION
**Pull request by dmikusa**
## Summary

- Removes `install-crane.sh`, `install-pack.sh` and `install-yj.sh`
- Switch from using custom install scripts to using `buildpacks/github-actions`
- Remove usage of deprecated packages like`ioutil`

## Use Cases

Pipeline clean up and simplification.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by dmikusa** [comment]
Tested with pipeline-builder-canary and all looks good.

File: cmd/statik/main.go [hunk 1]
```diff
@@ -16,11 +16,11 @@ import (
 // This can be useful to see what has changed between two versions.
 //
 // For example:
-//   1. `git co <branch>`
```
Comment (anthonydahanne) [inline cmd/statik/main.go:19]: unnecessary reformat ? 🤷

File: octo/builder_dependencies.go [hunk 2]
```diff
@@ -94,14 +94,11 @@ func contributeBuildImage(descriptor Descriptor, image string, classifier string
...
 					{
-						Name: "Install crane",
-						Run:  StatikString("/install-crane.sh"),
-						Env:  map[string]string{"CRANE_VERSION": CraneVersion},
-					},
-					{
-						Name: "Install yj",
-						Run:  StatikString("/install-yj.sh"),
-						Env:  map[string]string{"YJ_VERSION": YJVersion},
+						Uses: fmt.Sprintf("buildpacks/github-actions/setup-tools@v%s", BuildpackActionsVersion),
```
Comment (anthonydahanne) [inline octo/builder_dependencies.go:97]: nice, is that already published?

File: buildpack/rewriteLayer.go [hunk 3]
```diff
@@ -88,7 +88,9 @@ func rewriteLayer(layer v1.Layer, oldID, newID, oldVersion, newVersion string) (
 		}
 	}
 
-	return tarball.LayerFromReader(b)
+	return tarball.LayerFromOpener(func() (io.ReadCloser, error) {
```
Comment (anthonydahanne) [inline buildpack/rewriteLayer.go:91]: was the API deprecated? is it 100% equivalent?

File: octo/builder_dependencies.go (hunk 2 above)
Comment (dmikusa) [reply octo/builder_dependencies.go:97]: Yes, it's upstream. Its the same code. The only downfall is that it's bundled together so you can't get just pack or just yj. In some cases we need both, but not always. It seems like a very small price to pay though as the install is very quick (a single curl).

File: cmd/statik/main.go (hunk 1 above)
Comment (dmikusa) [reply cmd/statik/main.go:19]: Ugh, yes. I missed that.

File: cmd/statik/main.go [hunk 4]
```diff
@@ -16,11 +16,11 @@ import (
...
-//   2. `go run cmd/statik/main.go > old.txt`
-//   3. `git co <other-branch>`
-//   4. `go run cmd/statik/main.go > new.txt`
-//   5. `diff -u old.txt new.txt`
+//  1. `git co <branch>`
+//  2. `go run cmd/statik/main.go > old.txt`
+//  3. `git co <other-branch>`
+//  4. `go run cmd/statik/main.go > new.txt`
+//  5. `diff -u old.txt new.txt`
```
Comment (dmikusa) [inline cmd/statik/main.go]:
```suggestion
//   1. `git co <branch>`
//   2. `go run cmd/statik/main.go > old.txt`
//   3. `git co <other-branch>`
//   4. `go run cmd/statik/main.go > new.txt`
//   5. `diff -u old.txt new.txt`
```

File: buildpack/rewriteLayer.go (hunk 3 above)
Comment (dmikusa) [reply buildpack/rewriteLayer.go:91]: Yes, it was deprecated. That's my understanding, that it's the same.

https://pkg.go.dev/github.com/google/go-containerregistry/pkg/v1/tarball#LayerFromReader

Comment (dmikusa) [reply buildpack/rewriteLayer.go:91]: Although, I didn't look at the full method to see if things could be rewritten around this API change. I was just looking for the smallest change set.

File: octo/builder_dependencies.go (hunk 2 above)
Comment (anthonydahanne) [reply octo/builder_dependencies.go:97]: yeah, I used to think about using `pack` own github action, but... you're right, it's just curl https://github.com/buildpacks/github-actions?tab=readme-ov-file#setup-pack-cli-action

**Review by anthonydahanne — APPROVED** [review]

**Pull request merged**